resolves #198 resolves docinfo head, header and footer

ggrossetie opened the pull request:
Inspired by https://github.com/asciidoctor/asciidoctor-bespoke/blob/master/templates/slim/document.html.slim


I've added an example to make sure that it's working and here the result:

![tweet](https://user-images.githubusercontent.com/333276/73469381-91433180-4386-11ea-9b3e-5e7eb3feffc8.png)

As suggested by @mojavelinux I've added a unique suffix but it's a breaking change because `docinfo.html` won't be resolved (it must be renamed to `docinfo-revealjs.html`)

resolves #198

ggrossetie committed:
resolves #198 resolves docinfo head, header and footer

obilodeau reviewed (approved):
After a careful review of this, including reading [bespoke's documentation](https://github.com/asciidoctor/asciidoctor-bespoke#supplemental-content) and [upstream API doc](https://github.com/asciidoctor/asciidoctor/blob/2acdcece395a8327373b5f7ebf01c71dcd80ec24/lib/asciidoctor/document.rb#L1023), I can say that I like it and it is going in.

I'll merge shortly but right before doing so, I'll add documentation and document the breaking behavior in the CHANGELOG.

obilodeau committed:
doc: docinfo support, borrowed from Asciidoctor Bespoke

obilodeau committed:
Added doctest to docinfo example

Tests are skipped for now because there seems to be a problem with Asciidoctor-Doctest and docinfo.
